src/metabase/query_processor/middleware/fetch_source_query.clj

```diff
@@ -154,6 +160,7 @@
          card
 
          persisted? (and persisted-info/*allow-persisted-substitution*
+                         (not (segmented-user?))
```

Comment:
The actual fix. Just don't swap in the persisted query if the user is segmented (sandboxed)